vslib/src/sai_vs_port.cpp

```diff
@@ -11,7 +12,82 @@ sai_status_t vs_clear_port_all_stats(
     return SAI_STATUS_NOT_IMPLEMENTED;
 }
 
-VS_GENERIC_QUAD(PORT,port);
+sai_status_t vs_create_port(
+            _Out_ sai_object_id_t *port_id,
+            _In_ sai_object_id_t switch_id,
+            _In_ uint32_t attr_count,
+            _In_ const sai_attribute_t *attr_list)
+{
+    MUTEX();
+    SWSS_LOG_ENTER();
+
+    sai_attribute_t attr;
+
+    /* create port */
+    CHECK_STATUS(meta_sai_create_oid((sai_object_type_t)SAI_OBJECT_TYPE_PORT,
+                port_id,switch_id,attr_count,attr_list,&vs_generic_create));
+
+    attr.id = SAI_PORT_ATTR_ADMIN_STATE;
+    attr.value.booldata = false;     /* default admin state is down as defined in SAI */
+
+    CHECK_STATUS(vs_generic_set(SAI_OBJECT_TYPE_PORT, *port_id, &attr));
+
+    /* create priority groups */
+    const uint32_t port_pgs_count = 8;
```

Comment:
PGS are already created in VS in each class vslib/src/sai_vs_switch_xx.cpp
please remove

Reply:
It just creates for default ports in sai_vs_switch_xx.cpp. For breakout port, it needs to remove port and create port dynamically. I add a test case for breakout port, and it needs this patch. I will move this code to sai_vs_switch_xx.cpp, and create new pull request. Thanks.